reactor-core/src/test/kotlin/reactor/core/publisher/MonoFunctionsTests.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2011-2017 Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.publisher
+
+import org.junit.Test
+import reactor.test.StepVerifier
+import reactor.test.publisher.TestPublisher
+
+class MonoFunctionsTests {
+
+    @Test
+    fun `whenComplete with void Publishers`() {
+        val publishers = Array(3) { TestPublisher.create<Void>() }
+        publishers.forEach { it.complete() }
+        StepVerifier.create(whenComplete(*publishers))
+                .verifyComplete()
+    }
+
+    @Test
+    fun `whenComplete with two Monos`() {
+        StepVerifier.create(whenComplete("foo1".toMono(), "foo2".toMono()))
+                .verifyComplete()
+    }
+
+    @Test
+    fun `zip with Monos and combinator`() {
+        StepVerifier.create(zip("foo1".toMono(), "foo2".toMono(), "foo3".toMono()) { it.joinToString() })
```

Comment:
could you please extract `zip()`? Block outside of the zip's parentheses makes it a bit hard to read (tbh I was like, why `.create` has a block as the last argument? 😅